Lib/test/test_zipfile/test_path.py

```diff
@@ -191,24 +180,21 @@ def test_open_encoding_errors(self):
             with self.assertRaises(UnicodeDecodeError):
                 f.read()
 
-    def test_encoding_warnings(self):
+    @unittest.skipIf(
+        not getattr(sys.flags, 'warn_default_encoding', 0),
+        "Requires warn_default_encoding",
```

Comment:
Good point. I did consider that as I was refactoring in the zipp project. I figured it would be okay based on:

- The zipp project does [run with that setting enabled](https://github.com/jaraco/zipp/blob/44bf8757105b6687db196feae8090c24f367cc7f/tox.ini#L10-L11), so unlikely to regress without someone noticing.
- It's conceivable that some environment in CPython runs with that setting.
- Even if not, it seem likely that at some point the setting will be enabled for CPython.
- It's always possible to manually run with that setting.

My motivation for changing it was to make the test easier to read, validate the concern directly, avoid validating a backport of `assert_python_ok`, and avoid handling the conditions where that variable and the warning class doesn't exist.

It feels like an anti-pattern to have each module that tests behavior interacting with I/O to have a separate test that opens a subprocess in a different context to test that mode. We've already spent an inordinate amount of time addressing PEP 597. I'd like to stop the bleeding and reduce the maintenance burden added by honoring it.

WDYT?

Reply:
if this were something more critical I'd be inclined to care, but this test is for a trivial "is the warning showing up in the right place" situation.  I guess this is no big deal.